src/queryResult/singletonStore.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+class SingletonStore {
```

Comment:
This looks generic.   This should either be moved outside of `/queryResult/` or renamed to be `QueryResultSingletonStore` so it's apparent that it's just for query result data.

Idea for the future: generic (abstract?) singleton store so that each use-case for a singleton store can have their separate data.